replica/replica.go

```
@@ -1115,10 +1157,13 @@ func (r *Replica) unmarshalFile(file string, obj interface{}) error {
 	if err != nil {
 		return err
 	}
-	defer f.Close()
 
-	dec := json.NewDecoder(f)
-	return dec.Decode(obj)
+	err = json.NewDecoder(f).Decode(obj)
+	if err != nil {
+		return err
```

f.close is missing here

---

why this change required?

---

We were ignoring error thrown by f.Close().